lib/doorman.ex

```diff
@@ -1,2 +1,48 @@
 defmodule Doorman do
+  @moduledoc """
+  Provides helpers that take advantage of the options configured in your config
+  files.
+  """
+
+  @doc """
+  Authenticates a user by their email and password. Returns true user is found
+  and the password is correct, otherwise false.
+
+  Requires `user_module`, `secure_with`, and `repo` to be configured via
+  `Mix.Config`.
+  """
+  def authenticate(email, password) do
+    user = repo_module.get_by(user_module, email: email)
+    user && authenticate_user(user, password)
+  end
+
+  @doc """
+  Authenticates a user. Returns true user is found and the password is
+  correct, otherwise false.
+
+  Requires `user_module`, `secure_with`, and `repo` to be configured via
+  `Mix.Config`.
+  """
+  def authenticate_user(user, password) do
+    auth_module.authenticate(user, password)
+  end
+
+  defp repo_module do
+    get_module(:repo)
+  end
+
+  defp user_module do
+    get_module(:user_module)
+  end
+
+  defp auth_module do
+    get_module(:secure_with)
+  end
+
+  defp get_module(name) do
+    case Application.get_env(:doorman, name) do
+      nil -> raise "You must add `#{Atom.to_string(name)}` to `doorman` in your config"
```

Comment:
The generic error is pretty good, but it might be useful to have more specific error like

```elixir
```